cmd/monaco/deploy/deploy.go

```diff
@@ -34,40 +34,35 @@ import (
 	"github.com/spf13/afero"
 )
 
-func deployConfigs(fs afero.Fs, manifestPath string, environmentGroup string, specificEnvironments []string, specificProjects []string, continueOnErr bool, dryRun bool) error {
+func deployConfigs(fs afero.Fs, manifestPath string, environmentGroups []string, specificEnvironments []string, specificProjects []string, continueOnErr bool, dryRun bool) error {
 	absManifestPath, err := absPath(manifestPath)
 	if err != nil {
 		return fmt.Errorf("error while finding absolute path for `%s`: %w", manifestPath, err)
 	}
-	loadedManifest, err := loadManifest(fs, absManifestPath)
+	loadedManifest, err := loadManifest(fs, absManifestPath, environmentGroups, specificEnvironments)
```

Comment:
nitpick: This commit does a lot more than allowing multiple groups to be defined. More notably it replaces the external environment filtering with the manifest loader filtering. 

If you want nice atomic commits I'd split into 1) use the manifestLoader filtering, 2) allow multiple groups.
If you don't care for that, update the message so it's clear from the commit message what changed

Reply:
rebased both deploy and delete commits